torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -569,18 +610,10 @@ def perfect_match_inputs(
                     f"Actual {torch_input_compatible_types} vs\n"
                     f"expected {allowed_types}\n"
                 )
+                # NOTE: This is still a candidate for nearest match, as it only mismatches on dtypes.
+                self._record_matching_score(function_inputs, function_attributes)
+                diagnostic.with_additional_message(f"match score: {self.match_score}\n")
                 return False
```

Comment:
we are short circuiting the diagnostic message for attr dtype mismatches, if there is input dtype mismatch...

Reply:
I use a flag `is_perfect_match` to postpone the return to let diagnostics catch all it needs.